### PR TITLE
Internet validation improvements

### PR DESCRIPTION
**Pull request by ignaciocervinoGlob**
### Description
Fixing internet connection validations.

#### Camera:
- [x] If user enters to the camera and has no internet connection, the dialog returns to main

#### Add form:
- [x] If the user enter to the add form without connection, the dialog returns to main
- [x] If a user lose connection while interacting with the add form, the magnifying glass icon and save button show an alert that keeps the user in the same form, so he can solve the internet issue there and continue when he has internet.

#### Update form:
- [x] If user enters to the update form without internet, the internet dialog return him to the asset list results.
- [x] Same validation as add form

#### Search dialog:
- [x] If user selects location filter without internet, it shows immediately an internet dialog that dismiss the alert
- [x] If user press search button and there is no internet, the dialog return him to the camera or assets list.

#### Alert dialog
| Add manually button | Save button |
|---------------|-------------|
| ezgif com-gif-maker (1) | ezgif com-gif-maker (2) |

| No internet magnifying glass  | Update form |
|---------------|-------------|
| ezgif com-gif-maker (3) | ezgif com-gif-maker (4) |  

| Location filter  | Search button |
|---------------|-------------|
| ezgif com-gif-maker (5) | https://user-images.githubusercontent.com/90403672/154999708-15076c47-67dd-474f-ae35-8d2847952382.mp4 |

### CODE REVIEW

File: app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt [hunk 1]
```diff
@@ -35,24 +35,41 @@ class AssetFormActivity : AppCompatActivity() {
...
+                    getString(R.string.fragment_no_internet_connection_description),
+                    getString(R.string.asset_form_dialog_button_ok_title)
+                )
+            }
+            AssetFormState.NO_INTERNET_RETURN_MAIN -> {
+                showSimpleAlertDialog(
+                    this,
+                    getString(R.string.fragment_no_internet_connection_description),
+                    getString(R.string.fragment_dialog_button_return_to_main),
+                    { this.onBackPressed() }
```
Comment (SartiMau) [inline app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt:64]: this should call the viewmodel and from the ViewModel set a state and the activity react to that change calling the onBackPressed()

File: app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt [hunk 2]
```diff
@@ -35,24 +35,41 @@ class AssetFormActivity : AppCompatActivity() {
...
+                )
+            }
+            AssetFormState.NO_INTERNET_RETURN_PREVIOUS_ACTIVITY -> {
                 showSimpleAlertDialog(
                     this,
                     getString(R.string.fragment_no_internet_connection_description),
                     getString(R.string.asset_form_dialog_button_ok_title),
-                    { this.onBackPressed() },
-                    false
+                    { finish() }
```
Comment (SartiMau) [inline app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt:72]: this should call the viewmodel and from the ViewModel set a state and the activity react to that change calling the finish()

File: app/src/main/java/com/mobiletandil/stockapp/fragment/SearchPickerFragment.kt [hunk 3]
```diff
@@ -147,7 +147,13 @@ class SearchPickerFragment : DialogFragment() {
                 override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
                 override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
                 override fun afterTextChanged(s: Editable) {
-                    if (context?.let { context -> GeneralUtils.isOnline(context) } == false) {
+                    if (autocompleteTextView.text.toString().equals(getString(R.string.location_string_for_field), true) &&
+                        context?.let { context ->
```
Comment (SartiMau) [inline app/src/main/java/com/mobiletandil/stockapp/fragment/SearchPickerFragment.kt:151]: please try to put all this let statement into a one-line for more readability

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt (hunk 1 above)
Comment (ignaciocervinoGlob) [reply app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt]: done

File: app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt (hunk 2 above)
Comment (ignaciocervinoGlob) [reply app/src/main/java/com/mobiletandil/stockapp/activity/AssetFormActivity.kt]: done

File: app/src/main/java/com/mobiletandil/stockapp/fragment/SearchPickerFragment.kt (hunk 3 above)
Comment (ignaciocervinoGlob) [reply app/src/main/java/com/mobiletandil/stockapp/fragment/SearchPickerFragment.kt]: done

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**